cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/transaction/stream/StreamAdminTest.java

```diff
@@ -287,6 +285,18 @@ public void testOwner() throws Exception {
       // expected
     }
 
+    // trying to create same stream with different owner should fail
+    properties.put(Constants.Security.PRINCIPAL, "someOtherUser/someHost@somekdc.net");
+    try {
+      streamAdmin.create(stream, properties);
+      Assert.fail("Should have failed to add the same stream with different owner");
+    } catch (IllegalArgumentException e) {
```

Comment:
unrelated: why does this throw IllegalArgumentException? It is the wrong exception. It should be a ConflictException, or AlreadyExistsException, or something like that. No need to fix in this PR.